app/models/adopter.rb

```diff
@@ -161,6 +167,11 @@ def chimp_subscribe
     self.is_subscribed = true
   end
 
+  def comments_and_audits_and_associated_audits
+    clean_comments = comments.to_a.delete_if { |obj| obj.id == nil }
```

Comment:
There is an extra comment with a nil id in here because of the comment.build earlier up in the view.  This is ugly and will need some cleaning up.